src/renderer/api/kube-api.ts

```diff
@@ -174,4 +175,38 @@ export class KubeApi<T extends KubeObject = any> {
   }
 }
 
+export class VersionedKubeApi<T extends KubeObject = any> extends KubeApi<T> {
```

Comment:
@ixrock Extended API works fine, however what do you think about putting `VersionedKubeApi` features right inside parent `KubeApi`?

Reply:
It should be extracted to separate file, e.g. `kube-api-versioned.ts` next to `kube-api.ts`

Reply:
Extracting to a separate file will be simple, however once I look at how the semver package works perhaps it might be cleaner inlining the capability.